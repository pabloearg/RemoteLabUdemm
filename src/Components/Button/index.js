// @flow
import React from 'react';
import {
  TouchableOpacity, StyleSheet
} from 'react-native';
import type { ViewStyle, TextStyle } from '../../styles/types';
import TextHeadings from '../TextHeadings/TextHeadings';

type Props = {
  backgroundColor?: string,
  style?: ViewStyle,
  onPress: () => any,
  children: any,
  textStyle?: TextStyle,
  text?: string
}

const Button = (props: Props) => {
  const { backgroundColor, style, } = props;
  const { textStyle, ...rest } = props;
  Button.defaultProps = {
    backgroundColor: undefined,
    textStyle: undefined,
    style: undefined,
    text: undefined,
  };
  return (
    <TouchableOpacity
      /* eslint-disable react/jsx-props-no-spreading */
      {...rest}
      style={[{ backgroundColor }, styles.Button, style]}
      onPress={props.onPress}
    >
      <TextHeadings
        type="normal"
        styleText={[styles.Text, textStyle,]}
      >
        {props.text ?? props.children}

      </TextHeadings>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  Button: {
    alignItems: 'center',
    borderRadius: 5
  },
  Text: {
    fontSize: 12
  }
});
