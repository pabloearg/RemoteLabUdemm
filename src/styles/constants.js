export const LOW_SIZE_PERCENTAGE = 7.46 / 100;
export const MEDIUM_SIZE_PERCENTAGE = LOW_SIZE_PERCENTAGE * 2;
export const HIGH_SIZE_PERCENTAGE = LOW_SIZE_PERCENTAGE * 3;
