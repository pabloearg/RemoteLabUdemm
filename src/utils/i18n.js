// @flow
import I18n from 'i18n-js';

import es from '../static/locale/locale-es';

I18n.defaultLocale = 'es';
I18n.locale = 'es';

I18n.translations = {
  es,
};

export default I18n;
