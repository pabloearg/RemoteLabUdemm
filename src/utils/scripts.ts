import moment from "moment"
import { v4 as uuidv4 } from 'uuid';
export const getAppointmentsArray = (experimentId: string) => {

  const appointments = []
  for (let i = 0; i < 48; i++) {
    for (let hour = 9; hour < 15; hour++) {
      for (let minutes = 0; minutes < 60; minutes += 30) {
        const date = moment().set("hours", 0).set("minutes", 0)
        date.add(i, "days")
        date.add(hour, "hours")
        date.add(minutes, "minutes")
        const id = uuidv4()
        const appointment = {
          uuid: id,
          day: date.format("MM-DD-YYYY"),
          hour: date.format("hhmm"),
          isTaken: "0",
          experimentId: experimentId,
          id: id,
        }
        appointments.push(appointment)
      }

    }
  }
  return appointments
  console.log("appointments: ", appointments)
}