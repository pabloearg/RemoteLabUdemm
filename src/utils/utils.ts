import moment from 'moment';
import { Appointment } from '../API';

moment.locale('es-mx');

export const waitMillis = (timeInMillis: number): Promise<any> => new Promise((resolve) => setTimeout(resolve, timeInMillis));

export const convertExperimentArray = (array: any[]) => {
  const newObj = {};
  array.forEach((obj) => {
    newObj[obj.uuid] = obj;
  });
  return newObj;
};

export const getHourFromAppointment = (appointment: Appointment) => `${appointment?.hour?.substr(0, 2)}:${appointment?.hour?.substr(2, 2)}`;

export const getFormatedDayFromAppointment = (appointment: Appointment) => {
  if (!appointment?.day) {
    return '';
  }
  const momentDay = moment(appointment.day);
  return momentDay.format('DD MMM yyyy');
};
