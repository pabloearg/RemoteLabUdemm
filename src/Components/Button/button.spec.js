import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '.';
import TextStrong from '../TextStrong';
import { ORANGE } from '../../utils/constants';


describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Primary', () => {
      const component = shallow(
        <Button
          onPress={() => {}}
          backgroundColor={ORANGE}
        >
          Continue
        </Button>
      );
      expect(toJson(component)).toMatchSnapshot('Primary button snapshot');
    });
    it('should match to snapshot - Secondary', () => {
      const component = shallow(
        <Button onPress={() => {}}>
          <TextStrong>Probando</TextStrong>
        </Button>
      );
      expect(toJson(component)).toMatchSnapshot('Secondary button snapshot');
    });
  });

  describe('Interaction', () => {
    describe('onPressHandler', () => {
      it('should call onPress', () => {
        const mockOnPress = jest.fn();
        const component = shallow(
          <Button onPress={mockOnPress}>Probando</Button>
        );
        component.props().onPress();

        expect(mockOnPress).toHaveBeenCalled();
        expect(mockOnPress).toHaveBeenCalledTimes(1);
      });
    });
  });
});
