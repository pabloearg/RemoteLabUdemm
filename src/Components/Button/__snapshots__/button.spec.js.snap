// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot - Primary: Primary button snapshot 1`] = `
<ForwardRef
  backgroundColor="#F3A802"
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F3A802",
      },
      Object {
        "alignItems": "center",
        "borderRadius": 5,
      },
      undefined,
    ]
  }
>
  <TextStrong
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        undefined,
      ]
    }
  >
    Continue
  </TextStrong>
</ForwardRef>
`;

exports[`Button Rendering should match to snapshot - Secondary: Secondary button snapshot 1`] = `
<ForwardRef
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": undefined,
      },
      Object {
        "alignItems": "center",
        "borderRadius": 5,
      },
      undefined,
    ]
  }
>
  <TextStrong
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        undefined,
      ]
    }
  >
    <TextStrong
      style={Object {}}
    >
      Probando
    </TextStrong>
  </TextStrong>
</ForwardRef>
`;
